cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go

```diff
@@ -209,7 +244,19 @@ func CleanDir(filePath string) error {
 			return err
 		}
 	}
-	return nil
+	return err
+}
+
+func checkPath(path string, dirsToBeRetain []string) (bool, bool) {
+	for _, retainPath := range dirsToBeRetain {
+		if strings.HasPrefix(retainPath, path) {
+			if strings.EqualFold(retainPath, path) {
+				return false, true
+			}
+			return true, false
+		}
+	}
+	return false, false
```

Comment:
how about matching full paths only instead of returning two bools?
feels like checkPath is not needed?

- we walk the tree of the kubelet dir
- if we find that a retain dir is a prefix of the current path we skip this dir

it could also do the following:
- walk the tree and write all directories to a []string
- remove retain dirs
- delete all other dirs from this []string